src/pages/studyView/charts/ChartContainer.tsx

```diff
@@ -354,6 +359,26 @@ export class ChartContainer extends React.Component<IChartContainerProps, {}> {
     get chart() {
         switch (this.chartType) {
             case ChartTypeEnum.PIE_CHART: {
+                let pieData = this.props.promise.result;
+
+                if (
+                    this.colorMap.size == 0 &&
+                    this.props.promise.result.length != 0
+                ) {
+                    this.props.promise.result.map((item: any) => {
+                        this.colorMap.set(
+                            this.props.title + '.' + item.value,
+                            item.color
+                        );
+                    });
+                } else if (this.colorMap.size > 0) {
+                    for (let i = 0; i < pieData.length; i++) {
+                        pieData[i].color = this.colorMap.get(
+                            this.props.title + '.' + pieData[i].value
```

Comment:
Let's use string template literals for future string concatenation.

Reply:
Also you need a method to generate the key so you would not need to duplicate the logics.